flytepropeller/pkg/controller/nodes/executor.go

```diff
@@ -297,6 +297,7 @@ func (c *recursiveNodeExecutor) handleDownstream(ctx context.Context, execContex
 				// If the failure policy allows other nodes to continue running, do not exit the loop,
 				// Keep track of the last failed state in the loop since it'll be the one to return.
 				// TODO: If multiple nodes fail (which this mode allows), consolidate/summarize failure states in one.
+				stateOnComplete.ResetError()
```

Comment:
This means we keep the error in the CR because it is necessary for reporting in admin events. Then once it's reported we delete it right? My intuition says this should be behind the same configuration flag as deleting all the metadata for terminal nodes in failed states. Thoughts?